### PR TITLE
Feature/operation mul

### PR DESCRIPTION
**Pull request by xukuanHIT**
mul operation

### CODE REVIEW

File: pypose/lietensor/operation.py [hunk 1]
```diff
@@ -328,4 +457,291 @@ def backward(ctx, grad_output):
...
+        return grad_input.squeeze(-2)
+
+
+class SO3_Act(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, X, p):
+        Xv, Xw = X[..., :3], X[..., 3:]
+        uv = torch.linalg.cross(Xv, p, dim=-1)
+        uv += uv
```
Comment (wang-chen) [inline pypose/lietensor/operation.py:469]: Why it is not `uv = 2 * torch.linalg.cross(Xv, p, dim=-1)`?

File: pypose/lietensor/operation.py [hunk 2]
```diff
@@ -328,4 +457,291 @@ def backward(ctx, grad_output):
...
+    x = x.tensor() if hasattr(x, 'ltype') else x
+    input, out_shape = broadcast_inputs(x, p)
+    if lid == 1:
+        out = SO3_Act.apply(*input)
+    elif lid == 2:
+        out = RxSO3_Act.apply(*input)
+    elif lid == 3:
+        out = SE3_Act.apply(*input)
+    elif lid == 4:
+        out = Sim3_Act.apply(*input)
```
Comment (wang-chen) [inline pypose/lietensor/operation.py:713]: Shall we distribute the `lid`-defined operations to each their corresponding classes, similar to what we did for `Exp` and `Log`, then `lid` won't be needed? We can remove the `lid` definition when all operations are defined within Python.

The same questions apply to `lietensor_act4` and `lietensor_mul` also.

File: test/lietensor/test_lietensor.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def forward(self, x):
     c = pp.randn_SO3(1,5)
 
     e = a * b
-    f = a * c
+    # f = a * c
```
Comment (wang-chen) [inline test/lietensor/test_lietensor.py:76]: Remove commented lines if not needed.

File: test/lietensor/test_lietensor.py [hunk 4]
```diff
@@ -89,9 +89,9 @@ def forward(self, x):
     t = I.Retr(a)
 
     I = pp.identity_SE3(3)
-    t = I.Retr(a)
+    # t = I.Retr(a)
```
Comment (wang-chen) [inline test/lietensor/test_lietensor.py:92]: Remove commented lines if not needed.

File: test/lietensor/test_lietensor.py [hunk 5]
```diff
@@ -89,9 +89,9 @@ def forward(self, x):
...
 
-    r = pp.Retr(I, a)
+    # r = pp.Retr(I, a)
```
Comment (wang-chen) [inline test/lietensor/test_lietensor.py:94]: Remove commented lines if not needed.

File: pypose/lietensor/operation.py [hunk 6]
```diff
@@ -328,4 +457,291 @@ def backward(ctx, grad_output):
...
+        X_grad = dq @ SO3_Act4_Jacobian(out)
+        p_grad = dq @ SO3_Matrix4x4(X)
+        return torch.cat((X_grad.squeeze(-2), zero), dim = -1), p_grad.squeeze(-2)
+
+
+class SE3_Act4(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, X, p):
+        out = torch.cat(((SO3_Act.apply(X[..., 3:], p[..., :3]) + X[..., :3] * p[..., 3:]), p[..., 3:]), dim=-1)
```
Comment (wang-chen) [inline pypose/lietensor/operation.py:564]: Too long, how about splitting into two lines, e.g.,
```
name = O3_Act.apply(X[..., 3:], p[..., :3]) + X[..., :3] * p[..., 3:]
out = torch.cat((name , p[..., 3:]), dim=-1)
```
or
```
name = O3_Act.apply(X[..., 3:], p[..., :3]) 
out = torch.cat((name + X[..., :3] * p[..., 3:], p[..., 3:]), dim=-1)
```
remember to give a reasonable `name` instead.

File: pypose/lietensor/operation.py [hunk 7]
```diff
@@ -328,4 +457,291 @@ def backward(ctx, grad_output):
...
+        X_grad = dq @ RxSO3_Act4_Jacobian(out)
+        p_grad = dq @ RxSO3_Matrix4x4(X)
+        return torch.cat((X_grad.squeeze(-2), zero), dim = -1), p_grad.squeeze(-2)
+
+
+class Sim3_Act4(torch.autograd.Function):
+
+    @staticmethod
+    def forward(ctx, X, p):
+        out = torch.cat(((RxSO3_Act.apply(X[..., 3:], p[..., :3]) + X[..., :3] * p[..., 3:]), p[..., 3:]), dim=-1)
```
Comment (wang-chen) [inline pypose/lietensor/operation.py:600]: Too long, how about splitting into two lines?
Similar to the above?

File: pypose/lietensor/lietensor.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 from .backends import adjT, jinvp, act3, act4, toMatrix
 from .basics import vec2skew, cumops, cummul, cumprod
 from .basics import cumops_, cummul_, cumprod_
-from .operation import RxSO3_Log, SE3_Log, SO3_Log, Sim3_Log, rxso3_Exp, se3_Exp, sim3_Exp, so3_Exp
+from .operation import RxSO3_Log, SE3_Log, SO3_Log, Sim3_Log, lietensor_act, lietensor_act4, rxso3_Exp, se3_Exp, sim3_Exp, so3_Exp, lietensor_mul
```
Comment (wang-chen) [inline pypose/lietensor/lietensor.py:10]: too long, maybe this is better?
```
from .operation import RxSO3_Log, SE3_Log, SO3_Log, Sim3_Log
from .operation import rxso3_Exp, se3_Exp, sim3_Exp, so3_Exp
from .operation import lietensor_act, lietensor_act4, lietensor_mul
```
BTW, we may want to define `RxSO3_act` instead of `lietensor_act`.

**Review by wang-chen — CHANGES_REQUESTED** [review]
